akka-projection-core/src/main/scala/akka/projection/internal/InternalProjectionState.scala

```diff
@@ -350,19 +356,20 @@ private[akka] abstract class InternalProjectionState[Offset, Envelope](
   private def stopHandlerOnTermination(
       src: Source[Done, NotUsed],
       handlerLifecycle: HandlerLifecycle): Source[Done, Future[Done]] = {
-    src.watchTermination() { (_, futDone) =>
-      handlerStrategy.recreateHandlerOnNextAccess()
-      futDone
-        .andThen(_ => handlerLifecycle.tryStop())
-        .andThen {
-          case Success(_) =>
-            statusObserver.stopped(projectionId)
-          case Failure(AbortProjectionException) =>
-            statusObserver.stopped(projectionId) // no restart
-          case Failure(exc) =>
-            Try(statusObserver.stopped(projectionId))
-            statusObserver.failed(projectionId, exc)
-        }
-    }
+    src
+      .watchTermination() { (_, futDone) =>
+        handlerStrategy.recreateHandlerOnNextAccess()
+        futDone
+          .andThen { case _ => handlerLifecycle.tryStop() }
+          .andThen {
+            case Success(_) =>
+              statusObserver.stopped(projectionId)
+            case Failure(AbortProjectionException) =>
+              statusObserver.stopped(projectionId) // no restart
+            case Failure(exc) =>
+              Try(statusObserver.stopped(projectionId))
+              statusObserver.failed(projectionId, exc)
+          }
+      }
```

Comment:
It's here, here we make the final `Source` to mat to `Future[Done]`. This wasn't visible before because the signature was `Source[Done, _]`